src/test/java/com/revature/controllertests/MetricsControllerTest.java

```diff
@@ -1,270 +1,182 @@
 package com.revature.controllertests;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.testng.Assert.assertEquals;
 
-import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
 
-import org.junit.jupiter.api.MethodOrderer;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.TestMethodOrder;
-import org.junit.jupiter.api.extension.ExtendWith;
-import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.mock.mockito.MockBean;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.revature.cmsforce.CMSforceApplication;
 import com.revature.controllers.MetricsController;
+import com.revature.entities.Link;
+import com.revature.entities.Module;
+import com.revature.entities.ReqLink;
+import com.revature.services.ContentService;
+import com.revature.services.ModuleService;
+import com.revature.services.SearchService;
+import com.revature.services.TimegraphService;
 import com.revature.util.MetricsData;
+import com.revature.util.TimeGraphData;
 
+@SpringBootTest(classes = CMSforceApplication.class)
+public class MetricsControllerTest extends AbstractTestNGSpringContextTests {
```

Comment:
Looks like there are a lot of removed tests here, was that intentional? Why were they removed?

Reply:
Yes. I replaced them per the comment on the class below:
* This test class fails to test the functionality of the controller,
 * not to mention the controller is nearly untestable at this point.
 * These tests should be deprecated and replaced when the controller
 * is fixed.

Reply:
got it, thanks for the clarification!